src/osd/scheduler/mClockScheduler.h

```diff
@@ -130,7 +131,13 @@ class mClockScheduler : public OpScheduler, md_config_obs_t {
     true,
     true,
     2>;
+  using priority_t = unsigned;
+  using SubQueue = std::map<priority_t,
+	std::list<OpSchedulerItem>,
+	std::greater<priority_t>>;
+  using SubQueueIter = SubQueue::iterator;
   mclock_queue_t scheduler;
+  SubQueue high_priority;
```

Comment:
```
  /**
   * high_priority
   *
   * Holds entries to be dequeued in strict order ahead of mclock.
   * Invariant: entries are never empty
   */
```

Reply:
Also, it seems to me like we can remove immediate and use a special priority value of std::numeric_limits<priority_t>::max().

Reply:
@athanatos that makes sense to me. Do you think I can do it as part of this change?